smallrye-reactive-messaging-provider/src/main/java/io/smallrye/reactive/messaging/assembly/CleanupContextPropagation.java

```diff
@@ -0,0 +1,28 @@
+package io.smallrye.reactive.messaging.assembly;
+
+import java.util.List;
+import java.util.concurrent.Executor;
+
+import javax.enterprise.context.ApplicationScoped;
+
+import org.eclipse.microprofile.context.ThreadContext;
+
+import io.smallrye.reactive.messaging.ChannelRegistar;
+import io.smallrye.reactive.messaging.ChannelRegistry;
+
+@ApplicationScoped
+public class CleanupContextPropagation implements AssemblyHook {
+    @Override
+    public Executor before(List<ChannelRegistar> registars,
+            ChannelRegistry registry) {
+
+        try {
+            CleanupContextPropagation.class.getClassLoader().loadClass("org.eclipse.microprofile.context.ThreadContext");
+            System.out.println("Context Propgation hook called and clearing context...");
+            return ThreadContext.builder().cleared(ThreadContext.ALL_REMAINING).build().currentContextExecutor();
```

Comment:
This seems to need to be
```suggestion
            return ThreadContext.builder().propagated().cleared(ThreadContext.ALL_REMAINING).build().currentContextExecutor();
```
or I end up with an error:
> java.lang.IllegalStateException: Cannot use ALL_REMAINING in more than one of propagated, cleared, unchanged

To be safe it is might be better to also call unchanged():
```suggestion
            return ThreadContext.builder().propagated().cleared(ThreadContext.ALL_REMAINING).build().currentContextExecutor();
```